brevets/templates/calc.html

```diff
@@ -0,0 +1,310 @@
+                        $.getJSON(RETRIEVE_URL /* Send no data */).done(
+                          function(data) {
+                             console.log(data);
+                              if (data.status == 1) {
+                                  console.log("Successfully fetched: ", data);
+                                  $("#brevet_dist_km").val(data.result.brevet_dist);
+                                  $("#begin_date").val(data.result.start_time);
+                                  var control_list = data.result.control_list;
+                                  console.log("Control_list: ", control_list);
+                                  // We need to know how many items are in the fetched list
+                                  var num_items = control_list.length;
+                                  console.log("Fetched: ", num_items);
+                                  var current_cl = $('.control').length;
+                                  console.log("Current: ", current_cl);
+                                  console.log("Fetched: ", num_items);
+                                  if (num_items > current_cl) {
+                                      for (i = current_cl; i < num_items; i++)
+                                          addRow();
+                                  }
+                                  else if (num_items < current_cl) {
+                                      for (i = current_cl - 1; i >= num_items; i--)
+                                          $('.control').eq(i).remove();
+                                  }
+
+                                  // For each row in the HTML table
+                                  $('.control').each(function(index) {
+                                      var ctrl_l = $(this); // This is our row
+                                      // Clear existing values
+                                      ctrl_l.find("input[name='miles']").val('');
+                                      ctrl_l.find("input[name='km']").val('');
+                                      ctrl_l.find("input[name='loc']").val('');
+                                      ctrl_l.find("input[name='open_t']").val('');
+                                      ctrl_l.find("input[name='close_t']").val('');
+                                      // Check if we're already done?
+                                      if (index < num_items) {
+                                          ctrl_l.find("input[name='miles']").val(control_list[index].miles);
+                                          ctrl_l.find("input[name='km']").val(control_list[index].km);
+                                          ctrl_l.find("input[name='loc']").val(control_list[index].loc);
+                                          ctrl_l.find("input[name='open_t']").val(control_list[index].open_t);
+                                          ctrl_l.find("input[name='close_t']").val(control_list[index].close_t);
```

Comment:
I noticed you might have mixed up the names here - you use "[name='loc']", open_t and close_t when they should be "location", "open", and "close" since these are their names in the HTML. 
```suggestion
                                          ctrl_l.find("input[name='location']").val(control_list[index].loc);
                                          ctrl_l.find("input[name='open']").val(control_list[index].open_t);
                                          ctrl_l.find("input[name='close']").val(control_list[index].close_t);
```

This should fix the location, open, and close not being filled in. Before, the location, open, and close were being loaded because you weren't actually clearing them in the code before this (that I suggest removing) because they also had this same typo.